### PR TITLE
fsmonitor/darwin: fix hangs for submodules

### PR DESCRIPTION
**Pull request by KojiNakamaru**
cc: Ramsay Jones <ramsay@ramsayjones.plus.com>

### CODE REVIEW

File: builtin/fsmonitor--daemon.c [hunk 1]
```diff
@@ -1311,9 +1311,15 @@ static int fsmonitor_run_daemon(void)
...
 	if (!is_directory(state.path_gitdir_watch.buf)) {
+		char *buf;
+		int len;
 		strbuf_reset(&state.path_gitdir_watch);
 		strbuf_addstr(&state.path_gitdir_watch,
 			      absolute_path(repo_get_git_dir(the_repository)));
+		buf = state.path_gitdir_watch.buf;
+		len = state.path_gitdir_watch.len;
+		if (len >= 2 && buf[len - 2] == '/' && buf[len - 1] == '.')
+			strbuf_setlen(&state.path_gitdir_watch, len - 2);
```
Comment (dscho) [inline builtin/fsmonitor--daemon.c:1322]: How about using [`strbuf_strip_suffix()`](https://github.com/git/git/blob/v2.46.2/strbuf.h#L532-L539) instead?

```c
strbuf_strip_suffix(&state.path_gitdir_watch, "/.");
```

Also, did you want to add tests?

**Review by dscho — COMMENTED** [review]
It's awesome to see someone working on FSMonitor fixes.

By the way, you can ignore [the `osx-*` failures](https://github.com/gitgitgadget/git/actions/runs/11082796968/job/30796281418?pr=1802#step:4:1994): They are happening due to [a known issue with libcurl v8.10.0](https://lore.kernel.org/git/ZuPKvYP9ZZ2mhb4m@pks.im/) and will go away once the `macos-13` runners are updated to include v8.10.1 instead ([I expected this to be happening on Sep 23rd or soon thereafter](https://github.com/git-for-windows/git/issues/5159#issuecomment-2364518920), but it seems that there are delays).

File: builtin/fsmonitor--daemon.c (hunk 1 above)
Comment (KojiNakamaru) [reply builtin/fsmonitor--daemon.c:1322]: Thank you! I'll update the commit.

**Pull request closed without merging**